hangups/http_utils.py

```diff
@@ -5,57 +5,126 @@
-def fetch(method, url, params=None, headers=None, cookies=None, data=None,
-          connector=None):
-    """Make an HTTP request.
+class ClientSession(aiohttp.ClientSession):
+    """Session to file http requests
 
-    If the request times out or a encounters a connection issue, it will be
-    retried MAX_RETRIES times before finally raising hangups.NetworkError.
-
-    Returns FetchResponse.
+    Args:
+        cookies: dict, initial cookies for authentication
+        proxy: string, proxy url used for the request
     """
-    logger.debug('Sending request %s %s:\n%r', method, url, data)
-    error_msg = None
-    for retry_num in range(MAX_RETRIES):
-        try:
-            res = yield from asyncio.wait_for(aiohttp.request(
-                method, url, params=params, headers=headers, cookies=cookies,
-                data=data, connector=connector
-            ), CONNECT_TIMEOUT)
-            body = yield from asyncio.wait_for(res.read(), REQUEST_TIMEOUT)
-            logger.debug('Received response %d %s:\n%r', res.status,
-                         res.reason, body)
-        except asyncio.TimeoutError:
-            error_msg = 'Request timed out'
-        except aiohttp.ServerDisconnectedError as e:
-            error_msg = 'Server disconnected error: {}'.format(e)
-        except aiohttp.ClientError as e:
-            error_msg = 'Request connection error: {}'.format(e)
+    def __init__(self, cookies=None, proxy=None):
+        self._proxy = proxy
+        super().__init__(cookies=cookies)
+
+    @property
+    def cookies(self):
+        """get all cookies of the session
+
+        Returns:
+            dict, cookie name as key and cookie value as data
+        """
+        return {cookie.key: cookie.value
+                for cookie in self._cookie_jar}
+
+    def _get_cookie(self, name):
+        """get a cookie or raise an error for a missing one
+
+        Args:
+            name: string, the requested cookie name
+
+        Returns:
+            string, requested cookie value
+
+        Raises:
+            KeyError: the requested cookie was not set by the server
+        """
+        for cookie in self._cookie_jar:
+            if cookie.key == name:
+                return cookie.value
+        raise KeyError("Cookie '{}' is required".format(name))
+
+    @asyncio.coroutine
+    def _request(self, method, url, **kwargs):
+        """perform a single http request with authorization header for google
+
+        Args:
+            method: string, HTTP request method
+            url: string, target URI
+            kwargs: dict, see ``aiohttp.ClientSession._request``
+
+        Returns:
+            aiohttp.ClientResponse instance
+
+        Raises:
+            see ``aiohttp.ClientSession._request``
+        """
+        # pylint:disable=arguments-differ
+        kwargs['headers'] = kwargs.get('headers') or {}
+        kwargs['headers'].update(
+            channel.get_authorization_headers(self._get_cookie('SAPISID')))
```

Comment:
Could this inadvertently include the authorization headers in a non-Google request if someone re-purposes this class?